python/GafferScene/PassWedge.py

```diff
@@ -0,0 +1,94 @@
+#  modification, are permitted provided that the following conditions are
+#  met:
+#
+#      * Redistributions of source code must retain the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer.
+#
+#      * Redistributions in binary form must reproduce the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer in the documentation and/or other materials provided with
+#        the distribution.
+#
+#      * Neither the name of John Haddon nor the names of
+#        any other contributors to this software may be used to endorse or
+#        promote products derived from this software without specific prior
+#        written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+#  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+#  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+#  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+#  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+#  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+#  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+#  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+#  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+#  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+#  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+##########################################################################
+
+import IECore
+
+import Gaffer
+import GafferDispatch
+import GafferScene
+
+class PassWedge( GafferDispatch.TaskContextProcessor ) :
+
+	def __init__( self, name = "PassWedge" ) :
+
+		GafferDispatch.TaskContextProcessor.__init__( self, name )
+
+		self["in"] = GafferScene.ScenePlug()
+		self["out"] = GafferScene.ScenePlug( direction = Gaffer.Plug.Direction.Out )
+		self["out"].setInput( self["in"] )
+		self["out"].setFlags( Gaffer.Plug.Flags.Serialisable, False )
+
+		self["__ContextQuery"] = Gaffer.ContextQuery()
+		self["__ContextQuery"].addQuery( Gaffer.IntPlug( "value", defaultValue = 1 ) )
+		self["__ContextQuery"].addQuery( Gaffer.StringPlug( "value", defaultValue = "" ) )
+		self["__ContextQuery"]["queries"][0]["name"].setValue( "frameRange:start" )
+		self["__ContextQuery"]["queries"][1]["name"].setValue( "pass" )
+
+		self["__TimeWarp"] = Gaffer.TimeWarp()
+		self["__TimeWarp"].setup( self["in"] )
+		self["__TimeWarp"]["in"].setInput( self["in"] )
+		self["__TimeWarp"]["speed"].setValue( 0 )
+		self["__TimeWarp"]["offset"].setInput( self["__ContextQuery"]["out"][0]["value"] )
+
+		self["__OptionQuery"] = GafferScene.OptionQuery()
+		self["__OptionQuery"]["scene"].setInput( self["__TimeWarp"]["out"] )
+		self["__OptionQuery"].addQuery( Gaffer.StringVectorDataPlug( "value", defaultValue = IECore.StringVectorData() ) )
```

Comment:
I _think_ we can omit the "value" argument here and below, since that will be done by `addQuery()`.

Reply:
I've improved that in 5f38896 and also simplified some redundant use of `defaultValue`.